### PR TITLE
Http factory

### PR DESCRIPTION
**Pull request by fengzai6**
#### 💻 变更类型 | Change Type

- [x] feat
- [ ] fix
- [x] refactor
- [ ] chore
- [ ] style
- [ ] docs
- [ ] test

#### 🔀 变更说明 | Description of Change

新增 `http-factory` 模块，提供可复用的 HTTP 客户端工厂函数，替代各 API 模块中重复的 axios 封装逻辑。

**核心能力：**

- **Token 自动注入** — 请求拦截器自动从 `getAccessToken` 获取 token 并注入请求头
- **主动刷新** — 返回 `AccessTokenDetail`（含 `expiresAt`）时，在 token 即将过期前异步触发刷新，不阻塞当前请求，默认提前 60s，可自定义 `refreshBufferMs`
- **被动刷新** — 收到 401 或业务响应命中时，自动刷新 token 并重放原请求
- **并发控制** — `TokenRefreshManager` 确保多个并发请求只触发一次刷新
- **业务错误归一化** — 统一处理业务失败响应，支持自定义判定和映射逻辑
- **向后兼容** — `getAccessToken` 可返回 `string` 或 `{ token, expiresAt, refreshBufferMs }`，旧用法不受影响

**其他变更：**

- 现有 API 模块（`account`、`auth`、`cat`、`group`、`permission`、`role`、`user`）迁移至 `new-http`
- 新增完整的单元测试和边界用例测试
- 新增 `http-demo.ts` 示例文件

#### 📝 补充信息 | Additional Information

**配置示例：**

```ts
const http = createHttpClient({
  axiosConfig: { baseURL: '/api' },
  getAccessToken: () => {
    const token = store.getState().jwtToken;
    if (!token?.accessToken) return null;
    return {
      token: token.accessToken,
      expiresAt: new Date(token.expiresAt),
      // refreshBufferMs: 60_000, // 可选，默认 1 分钟
    };
  },
  refreshAccessToken: () => refreshToken(),
  onAuthFailure: () => store.getState().logout(),
});
```

#### 🔗 关联 Issue | Related Issue

N/A

## Summary by CodeRabbit

* **New Features**
  * Added real-time communication support (socket.io-client).

* **Chores**
  * Upgraded axios and added Vitest scripts (type-check, test, test:watch).
  * Introduced a new token-aware HTTP client with automatic token refresh and improved auth failure handling.
  * Migrated API callers to use the new HTTP client and adjusted TypeScript compiler options.

* **Documentation**
  * Added comprehensive README and a browser demo for the HTTP client.

* **Tests**
  * Added extensive tests and test utilities covering token refresh and edge cases.

### CODE REVIEW

File: apps/web/src/services/api/http-factory/http-demo.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  refreshAccessToken: async () => {
+    const refreshToken = browserTokenStore.getRefreshToken();
+
+    if (!refreshToken) {
+      throw new Error("refreshToken 不存在");
+    }
+
+    const response = await axios.post<unknown>("/auth/refresh", {
+      refreshToken,
+    });
```
[the remark this answers is not in the document]
Comment (fengzai6) [reply apps/web/src/services/api/http-factory/http-demo.ts:76]: 符合预期
